velox/docs/develop/timestamp.rst

```diff
@@ -0,0 +1,301 @@
+efficiently done by just overwriting the 12 bits, and that comparisons can be
+done by just comparing the 52 bits relative to timestamp (ignoring the timezone
+ID).
+
+However, unpacking/converting a TimestampWithTimezone into an absolute time
+definition requires a
+`timezone conversion <https://github.com/facebookincubator/velox/blob/main/velox/functions/prestosql/DateTimeFunctions.h#L74-L84>`_. 
+
+Conversions Across Timezones
+----------------------------
+
+A common operation required when processing timestamps and timezone is finding
+the wall clock and calendar read in a specific timezone given an absolute point
+in time described by a wall clock and calendar read in a different timezone.
+For example, at the exact point in time when UTC hits ``1970-01-01 00:00:00``,
+what was the wall clock read in China?
+
+Timezone conversions are tricky since they are non-linear and depend on
+daylight savings time schedules and other local regulations, and these change
+over time. To enable such conversions, `IANA <https://www.iana.org/time-zones>`_
+periodically publishes a global source of authority database for timezone
+conversions, which is periodically pushed to systems using packages like tzdata
+for Linux. 
+
+In Velox, Timezone conversions are done using std::chrono. Starting in C++20,
+std::chrono `supports conversion of timestamp across timezones <https://en.cppreference.com/w/cpp/chrono/time_zone>`_.
+To support older versions of the C++ standard, in Velox we vendor an
+implementation of this API at `velox/external/date/ <https://github.com/facebookincubator/velox/tree/main/velox/external/date>`_.
+This class handles timezone conversions by leveraging APIs provided by the
+operating system, based on the tzdata database installed locally. If systems
+happen to have inconsistent or older versions of the tzdata database, Velox’s
+conversions may produce inconsistent results. 
+
+You can check the tzdata installed in your system by:
+
+.. code-block:: bash
+
+  $ rpm -qa | grep tzdata
+  tzdata-2024a-1.fc38.noarch
+
+Timezone conversions are done using special methods in the Timestamp class:
+``Timestamp::toGMT()`` and ``Timestamp::toTimezone()``. They can take either a
+timezone ID or a date::time_zone pointer. Providing a date::time_zone is
+generally more efficient, but std::chrono does not handle time zone offsets
+such as ``+09:00``.  Timezone offsets are only supported in the API version
+that takes a timezone ID.
+
+Casts
+-----
+
+This section describes examples of timestamp casts following ANSI SQL
+semantics, using `Presto as a reference implementation <https://prestodb.io/docs/current/functions/datetime.html>`_,
+using ``set session legacy_timestamp = false;`` (see the section below for
+details).
+
+Timestamp literals are created based on whether time zone information is found
+on the string on not:
+
+.. code-block:: sql
+
+  presto:de> select typeof(TIMESTAMP '1970-01-01 00:00:00');
```

Comment:
Let's omit the `presto:de` part? Should we keep the format of the examples consistent as https://facebookincubator.github.io/velox/functions/presto/conversion.html#cast-to-timestamp

Reply:
Nice! That's better